parsl/db_logger.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+try:
+    from cmreslogging.handlers import CMRESHandler
+except ImportError:
+    _es_logging_enabled = False
+else:
+    _es_logging_enabled = True
+
+
+class OptionalModuleMissing(Exception):
+    ''' Error raised a required module is missing for a optional/extra provider
+    '''
+
+    def __init__(self, module_names, reason):
+        self.module_names = module_names
+        self.reason = reason
+
+    def __repr__(self):
+        return "Unable to Initialize logger.Missing:{0},  Reason:{1}".format(
+            self.module_names, self.reason
+        )
+
+
+class NullHandler(logging.Handler):
+    """Setup default logging to /dev/null since this is library."""
+
+    def emit(self, record):
+        pass
+
+
+def get_db_logger(host='search-parsl-logging-test-2yjkk2wuoxukk2wdpiicl7mcrm.us-east-1.es.amazonaws.com',
+                  port=443,
+                  level=logging.CRITICAL,
+                  index_name="parsl.campaign"):
+    """
+    Parameters
+    ----------
+    host : str, optional
+        URL to the elasticsearch cluster. Skip the http(s)://
+    port : int, optional
+        Port to use to access the elasticsearch cluster
+    level : logging.LEVEL, optional
+        This module will log *only* if logging.INFO or lower is provided.
```

Comment:
I don't think a configurable logging level is needed here. In traditional logging, it makes sense to control the verbosity. In the Elasticsearch case, we want everything to go to the DB. I think decisions about how to display info to the user should be done at the Kibana level.

Reply:
Sorry this isn't very clear. This logger only logs to Elasticsearch when the level is set to INFO or lower. This is not a traditional logger, and the logging level results in binary behavior.
```
if LEVEL > logging.INFO:
      return NullLogger
else
      return cmreslogger
```

Reply:
Right, since the purpose of the level is switching on whether or not we enable the cmsrellogger, why not just have the adjustable thing be a bool `enable_es_logging` instead of a configurable logging level? Then the purpose of the knob is stated explicitly in the name.

Reply:
What is the point of logging levels in the context of `cmreslogger`? I think we want everything to be sent-- we should always use the same logging level.